### DISCUSSION TITLE
unify with os-js?

### DISCUSSION THREAD

**Discussion by grue-bleen** [DustinBrett/daedalOS#144] [Ideas]
love the work done here, and the user-experience of [daedalOS](https://github.com/DustinBrett/daedalOS) is simply awesome.
however, i also love the work being done at [OS.js](https://github.com/os-js/OS.js), mainly the structure, application packaging, etc.
from a community perspective, wouldn't it be awesome if we could all work together and build the best of both worlds?

from a neutral community member's perspective (unaffiliated with either project at the moment), i would love to see daedalOS components and superior look-and-feel contributed into the OS.js structure and packaging model.  that would make for one ass-kicking web-desktop architecture.

see also: https://github.com/os-js/OS.js/issues/800

**Comment by DustinBrett** [comment]
Thanks! That's a cool idea indeed. OS.js was one of a few projects that inspired me early on in my work on my project.

I do indeed have a plan to make it easier to add 3rd party support to daedalOS, and it would be good if existing OS.js apps/packages had a way to run in daedalOS. Do you have any examples apps worth trying to port as a first attempt? Or any docs about their structure/package model?

**Reply by grue-bleen** [reply]
certainly.  take https://github.com/os-js/osjs-pdfreader-application as an "example" application.  it gets packaged in regular npm ways as @osjs/pdfreader-application, and includes [metadata](https://manual.os-js.org/tutorial/application/#metadata) that os.js can understand and deal with...

**Comment by DustinBrett** [comment]
Thanks for the example. Looking at https://github.com/os-js/osjs-pdfreader-application/blob/master/index.js it looks quite tricky. It would be nice if the osjs imports could be skipped and there was an option to just pass a canvas to my window structure. I'll continue to look into this but the packages don't look easy to have them work outside osjs infra/framework.

**Comment by grue-bleen** [comment]
thanks for looking into, dustin.  would you consider raising an issue on os-js project with your thoughts and recommendations?

**Comment by DustinBrett** [comment]
Np. I am not sure I want to push forward on that until I have a good suggestion for how the standard should be which I hope to have eventually when I add third party app support and figure out how that will look. At that time I may indeed suggest it in their project, but the way there code is I think it would be a lot of work to convert anything existing. Maybe we could all agree on something going forward for future apps.

**Reply by tmsteph** [reply]
https://manual.os-js.org/development/#developing-modules

What do you think of this? 

Looks like you can add modules in os-js via npm Packages.

**Reply by tmsteph** [reply]
https://manual.os-js.org/tutorial/application/

This also looks interesting.. The inside of an os-js "Application" and more!

**Reply by DustinBrett** [reply]
Ya it's interesting. I will look into it thanks! My OS is quite a bit differently structured and ideally I want something that doesn't need to rely on iframes, but otherwise I'd be happy to make something compatible with OS.js. I am not sure if they have many apps or a library/store type of thing.

**Reply by tmsteph** [reply]
Here are the Default "Extensions": 
https://manual.os-js.org/resource/official/

And these are some Community-Contributed  "Extensions:" 
https://manual.os-js.org/resource/contrib/

Looks a bit bare-bones but enough to get up and running. 

I'll think about writing some kind of compatibility layer, so that DaedalOS can run OS.js apps.

**Reply by DustinBrett** [reply]
Do you have a good use case for an app that they have which is worth porting over? I'd be more interested in bringing over the apps if I thought they were worth bringing.

**Reply by tmsteph** [reply]
I'm interested in how it all fits together and developing my own flavor!  

I like how easy and well documented the app-making process is on OS.js, and some of the vnc and other capabilities, connection with databases, ect. 

But I feel DaedalOS is much more snappy, responsive, and well-designed from a ux perspective. 

Eventually, I want to work almost completely through a browserOS, so that I have a portable distributed OS via my website no matter what device I'm on. 

My thought right now is to extend, embrace, and develop on top of existing solutions. 

So far I forked DeadalOS, and want to develop something called TommyOS. 

https://tmsteph.com  

Let me know if there are any issues with using the software in this way! 

Thanks, You are an inspiration! 

TMS

**Reply by DustinBrett** [reply]
Awesome, good luck! Sounds cool.
